src/S3/MultipartCopy.php

```diff
@@ -134,18 +135,25 @@ private function createPart($partNumber, $partsCount)
         // if the key contains a '?' character to specify where the query parameters start
         if (is_array($this->source)) {
             $key = str_replace('%2F', '/', rawurlencode($this->source['source_key']));
-            $data['CopySource'] = '/' . $this->source['source_bucket'] . '/' . $key;
+            $bucket = $this->source['source_bucket'];
         } else {
-
             list($bucket, $key) = explode('/', ltrim($this->source, '/'), 2);
-            $data['CopySource'] = '/' . $bucket . '/' . implode(
-            '/',
-            array_map(
-                'urlencode',
-                explode('/', rawurldecode($key))
-            )
-        );
+            $key = implode(
+                '/',
+                array_map(
+                    'urlencode',
+                    explode('/', rawurldecode($key))
+                )
+            );
         }
+
+        $uri = '/';
+        if (ArnParser::isArn($bucket)) {
```

Comment:
this could be condensed into one line:
 $uri = ArnParser::isArn($bucket) ? '' : "/";